src/components/sidebar/routePathView/LinkListView.tsx

```diff
@@ -0,0 +1,195 @@
+        this.setState({
+            selectedRoutePathLink: id,
+        });
+    }
+
+    private getRoutePathLinks = () => {
+        const routePathLinks = this.props.routePath.routePathLinks;
+        if (!routePathLinks) return;
+        return routePathLinks.map((routePathLink) => {
+            if (routePathLink.startNodeType === this.state.linkTableFilter ||
+                this.state.linkTableFilter === filterTypes.ALL) {
+                return(
+                    <div
+                        key={routePathLink.id}
+                        className={
+                            (this.state.selectedRoutePathLink === routePathLink.id)
+                            ? s.routePathLinkRowSelected : s.routePathLinkRow}
+                        onClick={this.selectRoutePathLink(routePathLink.id)}
+                    >
+                        <div className={s.flexInnerRow}>
+                            <div className={s.tableItemMedium}>
+                                {routePathLink.id}
+                            </div>
+
+                            <div className={s.tableItem}>
+                                {this.getNodeName(routePathLink.startNode)}
+                            </div>
+
+                            <div className={s.tableItem}>
+                                {this.getNodeName(routePathLink.endNode)}
+                            </div>
+
+                            <div className={s.tableItem}>
+                                {this.getHastusId(routePathLink)}
+                            </div>
+
+                            <div className={s.tableItemSmall}>
+                                {routePathLink.startNodeType}
+                            </div>
+                        </div>
+                    </div>);
+            }
+            return;
+        });
+    }
+
+    public getHastusId = (routePathLink: IRoutePathLink) => {
+        return (routePathLink.startNode.stop && routePathLink.startNode.stop.hastusId) ?
+            routePathLink.startNode.stop.hastusId :
+            '-';
+    }
+
+    public getNodeName = (node: INode) => {
+        return (node.stop) ?
+            node.stop.nameFi :
+            '-';
+    }
+
+    public openLinkView = () => {
+        if (this.state.selectedRoutePathLink) {
+            this.props.sidebarStore!.setOpenLinkId(parseInt(this.state.selectedRoutePathLink, 10));
+            const linkViewLink = routeBuilder.to(subSites.link).toLink();
+            window.open(linkViewLink);
+        }
+    }
+
+    public getLinkViewButtonTitle = () => {
+        const routePathLinkId = this.state.selectedRoutePathLink;
+        if (routePathLinkId) {
+            return `Avaa linkki (id: ${this.state.selectedRoutePathLink})`;
+        }
+        return `Ei valittua linkkiä`;
+    }
+
+    public isRoutePathLinkSelected = () => {
+        const routePathLinkSelected = this.state.selectedRoutePathLink;
+        if (routePathLinkSelected) return false;
```

Comment:
This function looks a bit illogical. We have isRoutePathLinkSelected() function that returns false if routePathLinkSelected if considered true.

Reply:
fixed